modules/core/src/main/java/org/apache/ignite/internal/client/thin/ClientUtils.java

```diff
@@ -121,119 +96,17 @@ static <E> void collection(E[] col, BinaryOutputStream out, BiConsumer<BinaryOut
         }
     }
 
-    /**
-     * @param in Input stream.
-     * @param elemReader Collection element deserializer.
-     * @return Deserialized collection.
-     */
-    static <E> Collection<E> collection(BinaryInputStream in, Function<BinaryInputStream, E> elemReader) {
-        Collection<E> col = new LinkedList<>(); // needs to be ordered for some use cases
-
-        int cnt = in.readInt();
-
-        for (int i = 0; i < cnt; i++)
-            col.add(elemReader.apply(in));
-
-        return col;
-    }
-
-    /**
-     * @return Deserialized map
-     */
-    private static <K, V> Map<K, V> map(
-        BinaryInputStream in,
-        Function<BinaryInputStream, K> keyReader,
-        Function<BinaryInputStream, V> valReader
-    ) {
-        int cnt = in.readInt();
-
-        Map<K, V> map = new HashMap<>(cnt);
-
-        for (int i = 0; i < cnt; i++)
-            map.put(keyReader.apply(in), valReader.apply(in));
-
-        return map;
-    }
-
     /** Deserialize binary type metadata from stream. */
     BinaryMetadata binaryMetadata(BinaryInputStream in) throws IOException {
```

Comment:
Let's move full method to BinaryUtils. Looks strange method that just calls BinaryUtils twice.